### ISSUE TITLE
The efficiency of synchronizing Pod status too slow.

### ISSUE THREAD

**Issue by FengyunPan2** [kubernetes/kubernetes#123875]
### What would you like to be added?

Use a single List api instead of multiple Get api.

### Why is this needed?

My k8s node is a big machine(500C CPU & 2000GB Memory), the maxPods of kubelet could be set to 200+.
When I delete 200+ pods or create 200+ pods, the synchronization of Pod status by the kubelet becomes too slow(about 230s).
Making a large number of GET API calls can indeed increase the load on the API server.

**Comment by ffromani** [comment]
do we have more precise numbers about the (in)efficiency? for example, a graph of pod count vs execution time

**Comment by matthyx** [comment]
> do we have more precise numbers about the (in)efficiency? for example, a graph of pod count vs execution time

I agree with @ffromani it would be nice to see how the new patch improves things.

**Comment by vaibhav2107** [comment]
/remove-lifecycle rotten
As PR is still open
